core/src/main/java/com/linecorp/armeria/server/ServerConfig.java

```diff
@@ -339,6 +349,20 @@ public boolean shutdownWorkerGroupOnStop() {
         return shutdownWorkerGroupOnStop;
     }
 
+    /**
+     * Returns the {@link ChannelOption}s and their values of {@link Server}'s server sockets.
+     */
+    public Map<ChannelOption<?>, Object> serverChannelOptions() {
+        return Collections.unmodifiableMap(serverChannelOptions);
+    }
+
+    /**
+     * Returns the {@link ChannelOption}s and their values of {@link Server}'s server child sockets.
+     */
+    public Map<ChannelOption<?>, Object> serverChannelChildOptions() {
+        return Collections.unmodifiableMap(serverChannelChildOptions);
```

Comment:
I think we can ust `ImmutableMap.copyOf()` and keep the immutable version as a member field.